packages/app/src/cli/models/app/error-parsing.ts

```diff
@@ -0,0 +1,103 @@
+
+    // Score based on error types - prioritize variants with fewer missing fields
+    let score = 0
+    let missingFieldCount = 0
+    let typeErrorCount = 0
+    let otherErrorCount = 0
+
+    for (const issue of variant.issues) {
+      if (issue.message?.includes('Required') || issue.message?.includes('required')) {
+        missingFieldCount++
+      } else if (issue.message?.includes('Expected') && issue.message?.includes('received')) {
+        typeErrorCount++
+      } else {
+        otherErrorCount++
+      }
+    }
+
+    // Scoring logic: prefer variants with fewer missing fields
+    // Missing fields are more likely to indicate the intended variant
+    // than type mismatches (which could indicate wrong variant entirely)
+    if (missingFieldCount > 0) {
+      // If there are missing fields, this could be the intended variant
+      score = 1000 - missingFieldCount * 10 - typeErrorCount - otherErrorCount
+    } else if (typeErrorCount > 0) {
+      // If only type errors, less likely to be the intended variant
+      score = 100 - typeErrorCount * 5 - otherErrorCount
+    } else {
+      // Other errors
+      score = 50 - otherErrorCount
+    }
+
+    if (score > bestScore) {
+      bestScore = score
+      bestVariant = variant
+    }
+  }
+
+  return bestVariant
+}
+
+export function parseHumanReadableError(issues: ExtendedZodIssue[]) {
+  let humanReadableError = ''
+
+  issues.forEach((issue) => {
+    // Handle union errors with smart variant detection
+    if (issue.code === 'invalid_union' && issue.unionErrors) {
+      // Find the variant that's most likely the intended one
+      const bestVariant = findBestMatchingVariant(issue.unionErrors)
+
+      if (bestVariant && bestVariant.issues && bestVariant.issues.length > 0) {
+        // Show errors only for the best matching variant
+        bestVariant.issues.forEach((nestedIssue) => {
+          const path = nestedIssue.path && nestedIssue.path.length > 0 ? nestedIssue.path.map(String).join('.') : 'root'
+          humanReadableError += `• [${path}]: ${nestedIssue.message}\n`
+        })
+      } else {
+        // Fallback to showing all variants if we can't determine the best one
+        humanReadableError += `• Configuration doesn't match any expected format:\n`
+        issue.unionErrors.forEach((unionError, index: number) => {
+          humanReadableError += `  Option ${index + 1}:\n`
+          if (unionError.issues) {
+            unionError.issues.forEach((nestedIssue) => {
+              const path =
+                nestedIssue.path && nestedIssue.path.length > 0 ? nestedIssue.path.map(String).join('.') : 'root'
+              humanReadableError += `    • [${path}]: ${nestedIssue.message}\n`
+            })
+          }
+        })
+      }
+    } else {
+      // Handle regular issues
+      const path = issue.path && issue.path.length > 0 ? issue.path.map(String).join('.') : 'root'
+      const message = issue.message ?? 'Unknown error'
```

Comment:
This piece of code appears 3 times in the function, it can be extracted to a helper function